File: finanze/domain/loan_calculator.py
from dataclasses import dataclass
from datetime import date
from typing import Optional

from domain.dezimal import Dezimal
from domain.global_position import InstallmentFrequency, InterestType


@dataclass
class LoanCalculationParams:
    loan_amount: Optional[Dezimal]
    interest_rate: Dezimal
    interest_type: InterestType
    euribor_rate: Optional[Dezimal]
    fixed_years: Optional[int]
    start: date
    end: date
    principal_outstanding: Optional[Dezimal]
    fixed_interest_rate: Optional[Dezimal] = None
    installment_frequency: InstallmentFrequency = InstallmentFrequency.MONTHLY


@dataclass
class LoanCalculationResult:
    current_installment_payment: Optional[Dezimal]
    current_installment_interests: Optional[Dezimal]
    principal_outstanding: Optional[Dezimal]
    installment_date: Optional[date] = None

File: mobile/src/domain/core/loanCalculator.ts
import { InterestType } from "./globalPosition"
import { Dezimal } from "@/domain/dezimal"

export interface LoanCalculationParams {
  loanAmount: Dezimal | null
  interestRate: Dezimal
  interestType: InterestType
  euriborRate: Dezimal | null
  fixedYears: number | null
  start: string
  end: string
  principalOutstanding: Dezimal | null
}

export interface LoanCalculationResult {
  currentMonthlyPayment: Dezimal | null
  currentMonthlyInterests: Dezimal | null
  principalOutstanding: Dezimal | null
  installmentDate?: string | null
}
